### PR TITLE
Feature/ha test update

### PR DESCRIPTION
**Pull request by gmaretic**
Just tweaked the tests to better reflect the state of the world

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1343)

### CODE REVIEW

File: atlasdb-cassandra-multinode-tests/src/test/java/com/palantir/cassandra/multinode/OneNodeDownMetadataTest.java [hunk 1]
```diff
@@ -51,7 +51,8 @@ public void putMetadataForTableThrows() {
         TableMetadata newTableMetadata = new TableMetadata(new NameMetadataDescription(),
                 new ColumnMetadataDescription(), ConflictHandler.IGNORE_ALL);
         assertThatThrownBy(() -> OneNodeDownTestSuite.db.putMetadataForTable(OneNodeDownTestSuite.TEST_TABLE,
-                newTableMetadata.persistToBytes())).isInstanceOf(IllegalStateException.class);
+                newTableMetadata.persistToBytes())).isExactlyInstanceOf(IllegalStateException.class)
+                .hasMessageContaining("At schema version UNREACHABLE");
```
Comment (harrybiddle) [inline atlasdb-cassandra-multinode-tests/src/test/java/com/palantir/cassandra/multinode/OneNodeDownMetadataTest.java]: Do we want to do any checking of what was/wasn't created here?

File: atlasdb-cassandra-multinode-tests/src/test/java/com/palantir/cassandra/multinode/OneNodeDownTableManipulationTest.java [hunk 2]
```diff
@@ -31,29 +32,44 @@
 
     @Test
     public void createTableThrows() {
+        assertThat(OneNodeDownTestSuite.db.getAllTableNames()).doesNotContain(NEW_TABLE);
         assertThatThrownBy(
                 () -> OneNodeDownTestSuite.db.createTable(NEW_TABLE, AtlasDbConstants.GENERIC_TABLE_METADATA))
-                .isExactlyInstanceOf(IllegalStateException.class);
+                .isExactlyInstanceOf(IllegalStateException.class)
+                .hasMessageContaining("At schema version UNREACHABLE");
+        // This documents and verifies the current behaviour, creating the table in spite of the exception
```
Comment (harrybiddle) [inline atlasdb-cassandra-multinode-tests/src/test/java/com/palantir/cassandra/multinode/OneNodeDownTableManipulationTest.java]: 👍

File: atlasdb-cassandra-multinode-tests/src/test/java/com/palantir/cassandra/multinode/OneNodeDownPutTest.java [hunk 3]
```diff
@@ -74,8 +74,8 @@ public void canPutUnlessExists() {
     @Test
     public void putUnlessExistsThrowsOnExists() {
         assertThatThrownBy(() -> OneNodeDownTestSuite.db.putUnlessExists(OneNodeDownTestSuite.TEST_TABLE,
-                ImmutableMap.of(OneNodeDownTestSuite.CELL_1_1, OneNodeDownTestSuite.DEFAULT_CONTENTS))).isInstanceOf(
-                KeyAlreadyExistsException.class);
+                ImmutableMap.of(OneNodeDownTestSuite.CELL_1_1, OneNodeDownTestSuite.DEFAULT_CONTENTS)))
+                .isInstanceOf(KeyAlreadyExistsException.class);
```
Comment (harrybiddle) [inline atlasdb-cassandra-multinode-tests/src/test/java/com/palantir/cassandra/multinode/OneNodeDownPutTest.java]: We could check here that the put failed

File: atlasdb-cassandra-multinode-tests/src/test/java/com/palantir/cassandra/multinode/OneNodeDownTableManipulationTest.java (hunk 2 above)
Comment (jboreiko) [inline atlasdb-cassandra-multinode-tests/src/test/java/com/palantir/cassandra/multinode/OneNodeDownTableManipulationTest.java]: Might want to note in the comment that this is an inconsistency in the PR.

**Comment by gmaretic** [comment]
Ready to merge

**Pull request merged**